internal/resources/common/configurationprofiles/constructors/helpers.go

```
@@ -0,0 +1,48 @@
+package constructors
+
+import (
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// convertToInt attempts robust conversion to int. Returns value and success bool.
+func ConvertToInt(val interface{}, fieldName string, index int) (int, bool) {
+	switch v := val.(type) {
+	case int:
+		return v, true
+	case float64:
+		intVal := int(v)
+		if float64(intVal) != v {
+			log.Printf("[WARN] Loss of precision converting float64 %f to int for %s ID at index %d. Using truncated value %d.", v, fieldName, index, intVal)
+		}
+		return intVal, true
+	case string:
+		intVal, err := strconv.Atoi(v)
+		if err != nil {
+			log.Printf("[WARN] Could not convert string '%s' to int for %s ID at index %d: %v. Skipping.", v, fieldName, index, err)
+			return 0, false
+		}
+		return intVal, true
+	default:
+		log.Printf("[WARN] Unexpected type %T for %s ID: %v at index %d. Skipping.", val, fieldName, val, index)
+		return 0, false
+	}
+}
+
+// getListFromSet is a helper function to safely extract a list of interfaces
+// from a *schema.Set stored within a map[string]interface{}.
+// It returns an empty slice if the key is not found or the value is not a *schema.Set.
+func GetListFromSet(data map[string]interface{}, key string) []interface{} {
```

This should probably return an error if it can't extract data, not an empty interface.

---

We could consider something like this, so to not make the caller pattern overkill:

func GetListFromSet(data map[string]interface{}, key string) ([]interface{}, error) {
    v, ok := data[key]
    if !ok || v == nil {
        // Not an error, just no data for this key
        return []interface{}{}, nil
    }
    
    set, ok := v.(*schema.Set)
    if !ok {
        // actual error - wrong type
        return nil, fmt.Errorf("value for key '%s' is not a *schema.Set, got %T", key, v)
    }
    
    if set == nil {
        // Also not an error, just empty
        return []interface{}{}, nil
    }
    
    return set.List(), nil
}

this would result in a caller pattern like

if computerIDsList, err := constructors.GetListFromSet(data, "computer_ids"); err != nil {
    return nil, fmt.Errorf("failed to get computer_ids: %w", err)
} else if len(computerIDsList) > 0 {
    scope.Computers = constructComputers(computerIDsList)
}

---

Can you format this in .md so I can read it in full?